plugin/diagnostics.py

```diff
@@ -43,7 +43,9 @@ def view_diagnostics(view: sublime.View) -> Dict[str, List[Diagnostic]]:
         file_name = view.file_name()
         if file_name:
             window_diagnostics = windows.lookup(view.window()).diagnostics.get()
-            return window_diagnostics.get(file_name, {})
+            for file in window_diagnostics:
```

Comment:
I'm not sure if it's wise to have a for-loop here as this is called mainly by filter_by_range which also loops over items. We might need a Path class with a custom equality operator and custom hash operator to stash it into dicts as keys. I'm guessing this is an unnoticable performance loss?

Reply:
I don't notice any performance loss caused by the for-loop and the function calls from `os.path`. But I agree it's not as clean as the `get` call or something similar. Another option could be to convert the drive letters in Windows paths to uppercase before storing the diagnostics. But that might be error prone because we would need to ensure to prevent doing that for Unix files and folders, where paths are case sensitive.